simpleui-server/src/simpleui-server.ts

```diff
@@ -286,6 +284,13 @@ export class SimpleUIServer {
                     const props = PropsFileReader.getProps(
                         `${req.params.propsStub}.properties`,
                         `${req.params.appName}`, cmdVars.webPort);
+
+                    ////// set up zmq sockets
+                    let zmq_ports_array = ServerUtil.getZMQPortsFromProps(props).map(p => Number(p));
+                    Logger.log(LogLevel.DEBUG, `ZMQ ports: ${zmq_ports_array}`);
+                    SuiData.zmqMap = new zmq_wrapper(zmq_ports_array);
+                    //////
+                    
```

Comment:
should this code be removed? it's also located at lines 175-182

Reply:
This was originally a copy paste error but now I think it should be there. Keeping this line would allow us to add or change zmq devices without having to restart the simpleui container. Is that too small of a use case to keep this code?

Reply:
My immediate thoughts:

- Is there any danger of leaking sockets doing it this way? 
  - What if someone calls this function with the same list of ports more than once? 
- Is there actually a provision for removing sockets?

Reply:
Now that I think of it, `zmq_ports_array` couldn't have a memory leak because it's reinitialized every time the props handler is called. Since `SuiData.zmqMap` is a Map, the only way it would grow in size is if someone were to add new zmq devices to `ui.properties`. If someone were to call the props handler multiple times and the same values were used, the `zmq_wrapper`'s would be overwritten by fresh instances of the class. I dont know if this would destroy the sockets of the original `zmq_wrapper` instances. I think, just to be safe, it'd be best to remove this piece of code. Especially since I dont know if the sockets would be destroyed if their `zmq_wrapper` instance is overwritten.